### PR TITLE
Unify nested and top level connection

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

This PR unifies nested and top level connections translation, so it is always generated the same way.

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4397.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 34 | 43 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 40 | 53 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 55 | 74 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 38 | 54 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2104 | 2153 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2217 | 2209 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 30 | 37 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 32 | 39 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2904 | 2790 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 38 | 53 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 65 | 133 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 121 | 169 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 67 | 64 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 155 | 174 | 3003 |
|  connections.Connection | 14049 | 14049 | 75 | 61 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 114 | 98 | 4516 |
|  create.SimpleMutation | 7 | 7 | 45 | 66 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 59 | 28 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 220 | 206 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 192 | 224 | 1179 |
|  2871.NestedRelationshipFilter | 23000 | 23081 | 97 | 68 | 4410 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 42 | 43 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 71 | 81 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 33 | 40 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 45 | 56 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 18 | 22 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 40 | 44 | 2174 |
|  query.QueryWhere | 9672 | 9684 | 30 | 39 | 2160 |
|  query.SimpleQueryWithNestedWhere | 9839 | 9851 | 41 | 49 | 2160 |
|  query.Nested | 10095502 | 10095502 | 7143 | 6888 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 7062 | 7086 | 2004000 |
|  query.OrFilterOnRelationships | 40795 | 42617 | 128 | 162 | 1292 |
|  query.OrFilterOnRelationshipsAndNested | 35141 | 35547 | 171 | 233 | 1292 |
|  query.QueryWithNestedIn | 13499 | 13331 | 55 | 49 | 1261 |
|  query.NestedConnectionWhere | 9801 | 9801 | 44 | 53 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9950 | 9860 | 60 | 109 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18890 | 18800 | 141 | 166 | 1630 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 145 | 181 | 1134 |
|  query.Fulltext | 64 | 64 | 30 | 32 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 36 | 44 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 56 | 59 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 59 | 71 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14209 | 14155 | 53 | 62 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14209 | 14155 | 100 | 76 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14990 | 14592 | 89 | 110 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14631 | 14437 | 89 | 114 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 93 | 63 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14419 | 14532 | 101 | 90 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 42 | 47 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 74 | 88 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 63 | 62 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 47 | 54 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 49 | 53 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 238 | 254 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 236 | 226 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 93 | 108 | 2002 |

</details>
        
Old Schema Generation: 24.713s
Schema Generation: 24.428s
Old Subgraph Schema Generation: 28.405s
Subgraph Schema Generation: 28.057s

File: packages/graphql/tests/tck/sort.test.ts [hunk 1]
```diff
@@ -19,26 +19,65 @@
...
 import { Neo4jGraphQL } from "../../src";
-import { formatCypher, translateQuery, formatParams } from "./utils/tck-test-utils";
+import { formatCypher, formatParams, translateQuery } from "./utils/tck-test-utils";
 
 describe("Cypher sort tests", () => {
     let typeDefs: string;
     let neoSchema: Neo4jGraphQL;
 
     beforeAll(() => {
+        // type Movie {
```
Comment (MacondoExpress) [inline packages/graphql/tests/tck/sort.test.ts:29]: Commented code

**Review by MacondoExpress — DISMISSED** [review]
LGTM, left some comments

File: packages/graphql/src/translate/queryAST/ast/operations/ConnectionReadOperation.ts [hunk 2]
```diff
@@ -111,305 +108,222 @@ export class ConnectionReadOperation extends Operation {
...
+        let withWhere: Cypher.With | undefined;
 
-            sortSubquery = this.getPaginationSubquery(nestedContext, edgesVar, paginationField);
-            sortSubquery.addColumns(totalCount);
+        if (authFilterSubqueries.length > 0) {
+            withWhere = new Cypher.With("*");
+            this.addFiltersToClause(withWhere, nestedContext);
+        } else {
+            this.addFiltersToClause(selectionClause, nestedContext);
         }
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/ConnectionReadOperation.ts:146]: These are not equivalent, `withWhere` is a clause that may or may not exists and needs to be added to the overall `concat`, which means that we cannot blindly use the `filterableClause` in your proposal or `selectionClause` will be used twice

I'm sure that this logic can be improved, but it is a non-trivial amount of extra work

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 3]
```diff
@@ -330,6 +330,14 @@ export class ReadOperation extends Operation {
         return new Cypher.Return([projection, returnVariable]);
     }
 
+    protected getReturnExpression(context: QueryASTContext): Cypher.Expr {
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts:333]: Good catch

File: packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeConnectionPartial.ts [hunk 4]
```diff
@@ -133,4 +133,28 @@ export class CompositeConnectionPartial extends ConnectionReadOperation {
     public addPagination(_pagination: Pagination): void {
         return undefined;
     }
+
+    private getSelectionClauses(
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeConnectionPartial.ts:137]: Actually, it is only used in two places, and just found out it is no longer used in ReadOperation, so I think for now we can keep this one here

**Review by MacondoExpress — APPROVED** [review]

**Pull request merged**